plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1419,3 +1422,249 @@ def synthetic_radiograph(
             intensity = -np.log10(intensity / I0)
 
     return h * u.m, v * u.m, intensity
+
+
+# *************************************************************************
+# Film Stack & Layers
+# *************************************************************************
+
+
+class Layer:
+    def __init__(self, thickness, energy_axis, stopping_power, active=True, name=""):
+        r"""
+        A layer in a detector film stack. The layer could either be an active
+        layer (the actual film medium) or an inactive layer (a filter or
+        inactive part of the film, such as a substrate.)
+
+        Tabulated stopping powers for protons and electrons can be found in the
+        `NIST PSTAR database
+        <https://physics.nist.gov/PhysRefData/Star/Text/PSTAR.html>`_
+        and the
+        `NIST ESTAR database
+        <https://physics.nist.gov/PhysRefData/Star/Text/ESTAR.html>`_.
+
+        Parameters
+        ----------
+
+        thickness : `~astropy.units.Quantity`
+            The thickness of the layer, in units convertible to meters.
+
+        energy_axis : `~astropy.units.Quantity`
+            The energies corresponding to the stopping power array.
+
+        stopping_power : `~astropy.units.Quantity`
+            The stopping power in the material, multiplied by the material
+            mass density. The stopping power is tabulated in units of
+            MeV cm\ :sup:`^2` / g, so this variable has units of MeV/cm.
+
+        active : `bool`, optional
+            If `True`, this layer is marked as an active layer. The default is `True`.
+
+        name : `str`, optional
+            An optional name for the layer.
+
+        """
+        self.thickness = thickness
+        self.energy_axis = energy_axis
+        self.stopping_power = stopping_power
+        self.active = active
+        self.name = name
+
+
+class Stack:
+    r"""
+    An ordered list of
+    `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects.
+
+
+    Parameters
+    ----------
+
+    layers : list of `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects
+        A list of the `~plasmapy.diagnostics.charged_particle_radiography.Layer` objects that make up the film stack.
+    """
+
+    def __init__(self, layers):
+        self._layers = layers
+        self._energy_bands = None
+
+    @property
+    def nlayers(self):
+        r"""
+        The number of layers in the stack.
+        """
+
+        return len(self._layers)
+
+    @property
+    def nactive(self):
+        r"""
+        The number of layers in the stack marked 'active'
+        """
+
+        return len([layer for layer in self._layers if layer.active])
+
+    @property
+    def thickness(self):
+        r"""
+        The total thickness of the stack.
+        """
+        thickness = np.array([layer.thickness.to(u.mm).value for layer in self._layers])
+        return np.sum(thickness) * u.mm
```

Comment:
We can probably skip the element-wise unit operations and operate on the whole array:
```suggestion
        thickness = u.Quantity([layer.thickness for layer in self._layers])
        return thickness.sum().to(u.mm)
```